### ISSUE TITLE
xformers attention to onnx?

### ISSUE THREAD

**Issue by hardlipay** [THUDM/CogVLM#26]
![1697164328632](https://github.com/THUDM/CogVLM/assets/126460983/aa944a5a-f05a-462d-b86f-a1ffc3a058a2)
requirements 里使用了xfomers，它是用来构建eva-clip的吗？
我之前做过eva转onnx的工作，如果是xformers构建的eva，其中有一个attention算子目前不能转成onnx，
而使用openclip-pytorch构建的eva是可以转为onnx模型的

请问，项目里的eva模型-e是指的具体哪一个模型，我暂时没有找到，我目前使用的是eva02-large-336，
可以直接用openclip-pytorch构建的eva模型来替换项目中的eva模型吗？

The requirements use xfomers, is it used to build eva-clip? I've worked with eva to onnx before, and if the eva is built with xformers, one of the attention operators can't be converted to onnx at the moment, whereas the eva built with openclip-pytorch can be converted to onnx models!

May I ask, which model does the eva model-e in the project refer to, I can't find it at the moment, and I'm currently using eva02-large-336, can I directly replace the eva model in the project with the openclip-pytorch built eva model?

**Comment by 1049451037** [comment]
We use the EVA02_CLIP_E_psz14_s4B: https://github.com/baaivision/EVA/tree/master/EVA-CLIP#eva-02-clip-series

**Comment by hardlipay** [comment]
> We use the EVA02_CLIP_E_psz14_s4B: https://github.com/baaivision/EVA/tree/master/EVA-CLIP#eva-02-clip-series

太感谢了，我之前忽略了这两个超大的eva，因为我看它们模型block叠了这么大还只在公开bench上取得了一丢丢的提升。
我最近测试了gpt4-v，eva02large和你们的项目cogvlm，举个例子：我用prompts或者QA任务去测试这三个模型是否认识交通场景中的红绿灯，只有cogvlm成功认出来，并且给了合理解释，所以我有两个疑惑。
首先，都是使用了eva，为什么我使用提示词不能分类？是不是eva的image-encoder是具有这个能力的，但是没有做过图文对齐。而cogvlm做了相关的对齐。
其次，如果认为是eva模型block和权重提升带来的优势，结合我之前看的一些文章和经验，是不是某种意义上讲，现在的一些公开测试集，并不能展现超大视觉模型所带来的的真实提升，就是说eva-e实际上要比eva-02-large强的多的多。
非常感谢你们的精彩工作和耐心解惑！

Thank you so much, I ignored these two oversized eva's before, because I see that they model block stacked so large and still only made a drop on the public bench. I recently tested gpt4-v, eva02large and your project cogvlm, as an example: I used prompts or QA tasks to test if these three models recognise traffic lights in a traffic scenario, and only cogvlm succeeded in recognising them and gave a reasonable explanation, so I'm confused in two ways. Firstly, all use eva, why can't I classify using cue words? Is it possible that eva's image-encoder is capable of this, but hasn't done the graphic alignment. And cogvlm did the relevant alignment. Secondly, if it is considered to be an advantage from the eva model block and weight boost, combined with some of the articles and experiences I've read before, isn't it in a sense that some of the public test sets nowadays don't show the real boost from the extra-large visual model, that is to say that eva-e is in fact much stronger than eva-02-large. Thank you very much for your wonderful work and patience in solving the problems!

**Comment by 1049451037** [comment]
Maybe it's because we also trained the eva parameters in the last several stages.

**Comment by Sleepychord** [comment]
@hardlipay There might be many reasons. First CogVLM additionally trains the vision expert (6B), which is large enough to contain new information beyond EVA-CLIP2-E. And it is possible that our new datasets may contains more signal lights than that of  EVA-CLIP2-E.

**Comment by hardlipay** [comment]
> @hardlipay There might be many reasons. First CogVLM additionally trains the vision expert (6B), which is large enough to contain new information beyond EVA-CLIP2-E. And it is possible that our new datasets may contains more signal lights than that of EVA-CLIP2-E.

get it ，thanks！

**Comment by ZihaoWang-CV** [comment]
> Maybe it's because we also trained the eva parameters in the last several stages.

![image](https://github.com/THUDM/CogVLM/assets/54048176/94fc4472-f698-48c0-a4ee-e28bb4f3d1be)


Could you please clarify at which stage the EVA-CLIP model is fine-tuned? The paper indicates that the vision encoder is not trainable during the SFT stage. 

Additionally, it appears that only three stages are mentioned in the paper: pre-training for captioning, pre-training for grounding, and SFT. May I ask what do you men for 'last several stages'? 

Thanks!
